client/src/components/publications/publicationsLayout/PublicationsEditor.js

```diff
@@ -0,0 +1,142 @@
+  )
+}
+`;
+
+export const EditorButton = styled.button`
+  color: white;
+  border: 1px solid #ccc;
+  cursor: pointer;
+  width: auto;
+  background-color: #007bff;
+  border-color: #007bff;
+  padding: .25rem .5rem;
+  font-size: .875rem;
+  line-height: 1.5;
+  border-radius: .2rem;
+  &:hover{
+    background-color: #0069d9;
+  }
+`;
+
+const PublicationsEditor = ({
+  options,
+  setOptions,
+  publication,
+  teamId,
+  sortPublications,
+  setShowCreateForm,
+  setShowImportForm,
+}) => {
+  const dispatch = useDispatch();
+
+  const handleUpdate = () => {
+    dispatch(updatePublicationOptions(teamId, options));
+  };
+  return (
+    <>
+      <StyledButtonGroup>
+        <Dropdown>
+          <Dropdown.Toggle size="sm">
+            Add
+          </Dropdown.Toggle>
+          <Dropdown.Menu>
+            <Dropdown.Item onClick={() => setShowCreateForm(true)}>Insert Publication</Dropdown.Item>
+            <Dropdown.Item onClick={() => setShowImportForm(true)}>Import Publications</Dropdown.Item>
+          </Dropdown.Menu>
+        </Dropdown>
+      </StyledButtonGroup>
+      <StyledButtonGroup>
+        Group By
+        {' '}
+        {Object.keys(layoutOptions).map((layout) => (
+          <ButtonGroupItem
+            press={options.layout === layoutOptions[layout]}
+            key={layout}
+            onClick={() => setOptions({ ...options, layout: layoutOptions[layout] })}
+          >
+            {layoutOptions[layout]}
+          </ButtonGroupItem>
+        ))}
+      </StyledButtonGroup>
+      <StyledButtonGroup>
+        Sort By
+        {' '}
+        {Object.keys(sortingOptions).map((sortBy) => (
+          <ButtonGroupItem
+            press={options.sortBy === sortingOptions[sortBy]}
+            key={sortBy}
+            value={sortingOptions[sortBy]}
+            onClick={(e) => {
+              setOptions({ ...options, sortBy: sortingOptions[sortBy] });
+              sortPublications(publication, e.target.value);
+            }}
+          >
+            {sortingOptions[sortBy]}
+          </ButtonGroupItem>
+        ))}
+        {options.layout === layoutOptions.BY_CATEGORY
+              && (
+              <ButtonGroupItem
+                press={options.sortBy === 'Category Title'}
+                value="Category Title"
+                onClick={(e) => {
+                  setOptions({ ...options, sortBy: e.target.value });
+                  sortPublications(publication, e.target.value);
+                }}
+              >
+                Category Title
+              </ButtonGroupItem>
+              )}
+      </StyledButtonGroup>
+      <StyledButtonGroup>
+        <EditorButton
+          onClick={handleUpdate}
+        >
+          Update Layout
+        </EditorButton>
+      </StyledButtonGroup>
+    </>
+  );
+};
+// props validation
+PublicationsEditor.propTypes = {
+  options: PropTypes.object.isRequired,
+  setOptions: PropTypes.func.isRequired,
+  publication: PropTypes.object.isRequired,
```

Comment:
I think publication is an array now? might need to change this validation.
![image](https://user-images.githubusercontent.com/48777884/133352190-520b433e-d4b5-41b6-8a1d-fbbbfe011b02.png)

Reply:
Thank you for that!